clients/cobol-lsp-vscode-extension/src/services/copybook/CopybookDownloadService.ts

```diff
@@ -12,552 +12,137 @@
-    element: CopybookProfile,
-    errors: Set<string>,
-    startTime: number,
-  ) {
-    {
-      await this.handleQueue(element, errors, progress);
-
-      if (this.queue.length === 0 && errors.size > 0) {
-        errors.clear();
-      }
-
-      if (this.queue.length === 0) {
-        TelemetryService.registerEvent(
-          "Download copybooks from MF",
-          ["copybook", "COBOL", experimentTag],
-          "total time to search copybooks on MF",
-          new Map().set(
-            "time elapsed",
-            TelemetryService.calculateTimeElapsed(startTime, Date.now()),
-          ) as unknown as TelemetryEventMeasurements,
-        );
-      }
-    }
-  }
-
-  private async run(
-    element: CopybookProfile,
-    errors: Set<string>,
-    startTime: number,
-  ) {
     await vscode.window.withProgress(
       {
         location: vscode.ProgressLocation.Notification,
         title: "Fetching copybooks",
       },
       async (
         progress: vscode.Progress<{ message?: string; increment?: number }>,
-      ) => this.process(progress, element, errors, startTime),
+      ) => {
+        await this.processCopybookDownload(
+          progress,
+          documentUri,
+          copybookNames,
+        );
+      },
     );
-    return startTime;
   }
 
-  private async handleQueue(
-    element: CopybookProfile,
-    errors: Set<string>,
+  private async processCopybookDownload(
     progress: vscode.Progress<{ message?: string; increment?: number }>,
-  ) {
-    const toDownload: CopybookProfile[] = [element];
-    while (this.queue.length > 0) {
-      const downloadItem = await this.queue.pop();
-      if (downloadItem) {
-        toDownload.push(downloadItem);
-      }
+    documentUri: string,
+    copybookNames: CopybookName[],
+  ): Promise<void> {
+    const totalCopybooksToDownload = copybookNames.length;
+    let processedCopybooks = 0;
+
+    await Promise.all(
+      copybookNames.map(async (copybookName) => {
+        await this.downloadResolver!.downloadCopybook(
+          copybookName,
+          documentUri,
+        ).finally(() => {
+          processedCopybooks++;
+          this.updateDownloadProgress(
+            progress,
+            totalCopybooksToDownload,
+            processedCopybooks,
+          );
+        });
+      }),
+    ).catch((err) => {
+      vscode.window.showErrorMessage(err.message);
+    });
+  }
+
+  private async isPrerequisiteForDownloadSatisfied(
+    documentUri: string,
+    copybookNames: CopybookName[],
+  ): Promise<boolean> {
+    if (
+      !DownloadUtil.areCopybookDownloadConfigurationsPresent(
+        documentUri,
+        copybookNames,
+      )
+    )
+      return false;
+    if (!this.explorerAPI) return false;
+    const profile = await ProfileUtils.getProfileNameForCopybook(documentUri);
+    const availableProfiles = ProfileUtils.getAvailableProfiles(
+      this.explorerAPI!,
+    );
+    if ((!profile || !availableProfiles.includes(profile)) && true) {
```

Comment:
```suggestion
    if (!profile || !availableProfiles.includes(profile)) {
```

Reply:
Thanks ! it was a typo while refactoring